modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/AssetLibraryResourceImpl.java

```diff
@@ -453,6 +511,34 @@ private Map<String, Boolean> _getDepotAppCustomizationMap(
 		return depotAppCustomizationMap;
 	}
 
+	private List<Long> _getEligibleGroupIds(Group group) throws Exception {
+		List<Long> ids = new ArrayList<>();
+
+		long groupId = group.getGroupId();
+
+		List<DepotEntry> depots =
+			_depotEntryService.getGroupConnectedDepotEntries(
+				groupId, -1, QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		for (DepotEntry depot : depots) {
+			Group depotGroup = groupLocalService.fetchGroup(depot.getGroupId());
+
+			if ((depotGroup != null) && _isTrashEnabled(depotGroup)) {
+				ids.add(depotGroup.getGroupId());
+			}
+		}
+
+		Group scopeGroup = groupLocalService.fetchGroup(groupId);
+
+		if ((scopeGroup != null) && scopeGroup.isDepot() &&
+			_isTrashEnabled(scopeGroup)) {
```

Comment:
Can you use `_trashHelper.isTrashEnabled(groupId)` here or do you need the information coming from the patch request (not saved yet)?